src/peer_manager.rs

```diff
@@ -1141,6 +1158,48 @@ impl PeerManager {
             .collect()
     }
 
+    /// Returns the `PeerId`s and names of all the peers held in the RT, and whether they are
+    /// connected via a tunnel or not.
+    pub fn get_routing_peer_details(&self) -> Vec<(PeerId, XorName, bool)> {
+        self.routing_table
+            .iter()
+            .filter_map(|name| -> Option<(PeerId, XorName, bool)> {
+                let peer = match self.peer_map.get_by_name(name) {
+                    Some(peer) => peer,
+                    None => {
+                        error!("{:?} has {} in RT, but has no entry in peer_map for it.",
+                               self,
+                               name);
+                        return None;
+                    }
+                };
+                let peer_id = match peer.peer_id {
+                    Some(peer_id) => peer_id,
+                    None => {
+                        error!("{:?} has {} in RT, but has no peer ID for it.", self, name);
+                        return None;
+                    }
+                };
+                let is_tunnel = match peer.state {
+                    PeerState::Routing(RoutingConnection::Tunnel) => true,
+                    PeerState::Routing(_) => false,
+                    _ => {
+                        error!("{:?} has {} in RT, but has state {:?} for it.",
+                               self,
+                               name,
+                               peer.state);
+                        return None;
```

Comment:
And again, though I'm less sure what should be done in this case (maybe still reject since it's in an errornous state). Maybe again it's not worth handling this stuff now.